ambientweather2sqlite/awparser.py

```
@@ -112,3 +114,97 @@ def extract_labels(html_content: str) -> dict[str, str]:
     parser.feed(html_content)
     return parser.data_dict
+
+
+class UnitsHTMLParser(HTMLParser):
+    """
+    A parser to extract selected weather station units from an HTML file.
+
+    This parser identifies sections for each unit type as defined in the Units
+    enum, finds the corresponding <select> element, and extracts the text from
+    the <option> tag that has the 'selected' attribute.
+    """
```

Ruff (0.11.9)

Multi-line docstring summary should start at the first line

Remove whitespace after opening quotes

(D212)